rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -242,12 +243,14 @@ non_defaulted_zero_initialized_members = [
 
   // setters for named parameter idiom
 @[for member in message.structure.members]@
+@[  if len(message.structure.members) != 1 or member.name != EMPTY_STRUCTURE_REQUIRED_MEMBER_NAME]@
```

Comment:
@dirk-thomas nit: I'd think you can do this check once before the loop.

Reply:
Done in c9aae0c181721e22ae59ddf9302907e98ff0b9e5.